src/Razor/src/Microsoft.VisualStudio.LanguageServices.Razor/Remote/RemoteServiceInvoker.cs

```diff
@@ -51,7 +51,7 @@ internal sealed class RemoteServiceInvoker(
     {
         var client = typeof(IRemoteJsonService).IsAssignableFrom(typeof(TService))
             ? await TryGetJsonClientAsync(cancellationToken).ConfigureAwait(false)
-            : await TryGetClientAsync(cancellationToken).ConfigureAwait(false);
+            : await TryGetClientAsync(doInit: true, cancellationToken).ConfigureAwait(false);
```

Comment:
Yes, it came out a bit convoluted. I believe I misunderstood the comment in the code that (to me) made it sound like you always had to get MessagePack client before JSON client. I believe it was because init service was getting called via MessagePack client, and since it's now getting called via JSON client, we no longer need to always get MessageP:ack client. @davidwengier does that sound right?

Reply:
I think it's all addressed in the last commit.